pkg/storage/ingest/reader.go

```diff
@@ -363,15 +363,15 @@ func (r *PartitionReader) notifyLastConsumedOffset(fetches kgo.Fetches) {
 	})
 }
 
-func (r *PartitionReader) recordFetchesMetrics(fetches kgo.Fetches) {
+func (r *PartitionReader) recordFetchesMetrics(fetches kgo.Fetches, state string) {
 	var (
 		now        = time.Now()
 		numRecords = 0
 	)
 
 	fetches.EachRecord(func(record *kgo.Record) {
 		numRecords++
-		r.metrics.receiveDelay.Observe(now.Sub(record.Timestamp).Seconds())
+		r.metrics.receiveDelay.WithLabelValues(state).Observe(now.Sub(record.Timestamp).Seconds())
```

Comment:
Calls to `WithLabelValues()` are expensive. Since this is called for each record, it could end up in the profile. Since we just have 2 states, WDYT about storing the 2 state metrics directly in `r.metrics` (e.g. `r.metrics.receiveDelayWhenStarting` and `r.metrics.receiveDelayWhenRunning`) and then using them here?

Reply:
Good idea, implemented.